src/components/ModelStateIndicator.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEngineCommands } from './EngineCommands'
+import { Spinner } from './Spinner'
+import { CustomIcon } from './CustomIcon'
+
+export const ModelStateIndicator = () => {
+  const [commands] = useEngineCommands()
+
+  const lastCommandType = commands[commands.length - 1]?.type
```

Comment:
This is a great idea, but if I'm not mistaken we actually receive multiple "done" messages at the moment, maybe exactly two? We should make that not the case, but I think we'll see this indicator flicker once accordingly on each load for this reason. Not a huge deal but would be a bigger deal if lots of tests relied on it to know they're good to advance.